src/main/java/grupopqc/utfprtd/hybridexample/Algorithms/KeyEstablishmentStrategy.java

```diff
@@ -1,41 +1,60 @@
-package grupopqc.utfprtd.hybridexample;
+package grupopqc.utfprtd.hybridexample.Algorithms;
+
+import org.bouncycastle.pqc.jcajce.spec.KyberParameterSpec;
 
 import java.security.KeyPair;
 import java.security.PrivateKey;
 import java.security.PublicKey;
 import java.util.Map;
+import java.util.Objects;
 
 /**
  *
  * Strategy Pattern for the key establishment modes
  */
 public interface KeyEstablishmentStrategy {
+
+    KyberParameterSpec kyberParameterSpec = KyberParameterSpec.kyber768;
+    String provaderClassicStrategy = "BC";
+    String algorithmClassicStrategy = "ECDH";
+    String pqcParameterSpecs = "KYBER768";
+    String classicParameterSpec = "P-384";
+    String providerName = "BCPQC";
+
+    byte[] ukm = new byte[32];
+
+    /**
+     *
+     * @param algorithm parameter used to define which encryption algorithm to use with its additional settings
+     * @return void
+     */
+    void setPqcParameterSpecs(String algorithm);
+
+    /**
+     *
+     * @param providerName specific provader parameter
+     */
+    void setProviderName(String providerName);
     /**
     * Generates KeyPair(s)
-    * @param algorithmSpec the name of the algorithm based on the desired crypto. provider
-    * @param providerName  the name of the (BC) provider.
     * @return              the map of keyPair(s) generated. The string part could be defined as "PQC", "Classical" for indexing modes.
-    */
-    Map<String, KeyPair> keyGeneration(String algorithmSpec, String providerName);   
+     * @return void
```

Comment:
I think it should be "the map of keyPair(s) generated. The string part could be defined as "PQC", "Classical" for indexing modes."